docs/src/developers/style.md

```diff
@@ -51,6 +51,220 @@ format("test")
 The following sections outline extra style guide points that are not fixed
 automatically by JuliaFormatter.
 
+### Abstract types and composition
+
+Specifying types for method arguments and struct fields is mostly optional in
```

Comment:
I wouldn't throw struct field types into the same baskets, it is a common misunderstanding for new users that typing arguments is optional but typing struct fields is important for performance